Cython/Compiler/Code.py

```diff
@@ -2415,22 +2415,16 @@ def get_cached_constants_writer(self, target=None):
         return self.globalstate.get_cached_constants_writer(target)
 
     def name_in_module_state(self, cname):
-        # If we don't have a funcstate scope assume global scope for now
-
-        # default fallback option if we don't know better
-        modulestate_name = Naming.modulestateglobal_cname
+        if self.funcstate.scope is None:
+            return self.name_in_main_c_code_module_state(cname)
+        return self.funcstate.scope.name_in_module_state(cname)
 
-        # self.funcstate can be None in a few places, including ToPyStructUtilityCode.
-        # TODO - this will eventually need to be fixed.
-        if self.funcstate is not None:
-            if self.funcstate.scope is None or self.funcstate.scope.is_module_scope:
-                modulestate_name = Naming.modulestatevalue_cname
-            # TODO - more choices depending on the type of env
-            # e.g. slot, function, method
-
-        return f"{modulestate_name}->{cname}"
+    @staticmethod
+    def name_in_main_c_code_module_state(cname):
+        return f"{Naming.modulestateglobal_cname}->{cname}"
```

Comment:
I think this can go into `ModuleScope.name_in_module_state()`.

Reply:
Don't think it can because the places where this is used are explicitly where a scope isn't available (at least some of the time... it's really inconsistent even within the same function if a scope is available).